src/button.scss

```diff
@@ -83,321 +78,295 @@ $block: #{$fd-namespace}-button;
+  @include standardPressedFocus();
 
-    &.#{$block}--ghost {
-      @include ghostPressed();
-      @include ghostPressedFocus();
-    }
+  &.#{$block}--ghost {
+    @include ghostPressed();
+    @include ghostPressedFocus();
+  }
 
-    &.#{$block}--positive {
-      @include positivePressed();
-      @include positivePressedFocus();
-    }
+  &.#{$block}--positive {
+    @include positivePressed();
+    @include positivePressedFocus();
+  }
 
-    &.#{$block}--negative {
-      @include negativePressed();
-      @include negativePressedFocus();
-    }
+  &.#{$block}--negative {
+    @include negativePressed();
+    @include negativePressedFocus();
+  }
 
-    &.#{$block}--attention {
-      @include attentionPressed();
-      @include attentionPressedFocus();
-    }
+  &.#{$block}--attention {
+    @include attentionPressed();
+    @include attentionPressedFocus();
+  }
 
-    &.#{$block}--emphasized {
-      @include emphasizedPressed();
-      @include emphasizedPressedFocus();
-    }
+  &.#{$block}--emphasized {
+    @include emphasizedPressed();
+    @include emphasizedPressedFocus();
   }
 }
 
 @mixin buttonContainerHover() {
-  @include iconOverwrite() {
-    @include standardHover();
+  @include standardHover();
+
+  &.#{$block}--ghost {
+    @include ghostHover();
+  }
+
+  &.#{$block}--positive {
+    @include positiveHover();
+  }
+
+  &.#{$block}--negative {
+    @include negativeHover();
+  }
 
-    &.#{$block}--ghost {
-      @include ghostHover();
+  &.#{$block}--attention {
+    @include attentionHover();
+  }
+
+  &.#{$block}--emphasized {
+    @include emphasizedHover();
+  }
+}
+
+@mixin iconStylesLtr() {
+  > [class*='sap-icon'] {
+    color: inherit;
+    font-style: normal;
+    width: $fd-button-icon-font-size;
+    height: $fd-button-icon-font-size;
+    font-size: $fd-button-icon-font-size;
+    line-height: $fd-button-icon-font-size;
+    vertical-align: text-bottom;
+
+    &::before,
+    &::after {
+      width: inherit;
+      height: inherit;
+      font-size: inherit;
+      line-height: inherit;
+    }
+
+    &.#{$block}__icon--left {
+      margin-right: $fd-button-icon-margin;
     }
 
-    &.#{$block}--positive {
-      @include positiveHover();
+    &.#{$block}__icon--right {
+      margin-left: $fd-button-icon-margin;
     }
 
-    &.#{$block}--negative {
-      @include negativeHover();
+    @include fd-only-child() {
+      &.#{$block}__icon--left,
+      &.#{$block}__icon--right {
+        margin: 0;
+      }
     }
+  }
+}
 
-    &.#{$block}--attention {
-      @include attentionHover();
+@mixin iconStylesRtl {
+  @include fd-rtl() {
+    > .#{$block}__icon--left {
+      margin-right: 0;
+      margin-left: $fd-button-icon-margin;
     }
 
-    &.#{$block}--emphasized {
-      @include emphasizedHover();
+    > .#{$block}__icon--right {
+      margin-left: 0;
+      margin-right: $fd-button-icon-margin;
     }
   }
 }
 
+@mixin iconStyles {
+  @include iconStylesLtr();
+  @include iconStylesRtl();
```

Comment:
I don't think RTL should be splitted from LTR like that. Using `@include iconStylesRtl();` is misleading. I'd expect it to bring full icon styling for RTL mode, but in fact this brings only a modifier and to properly work it musts always be used together with LTR mixin.

Reply:
Great Idea, it really can be misleading